Build iot device page on /partners/iot-device

minkyngkm opened the pull request:
## Done

- Build a new page on `/partners/iot-device`

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8002/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Please refer to this [brief](https://docs.google.com/document/d/1vOIffTaJL8LeQVj-kuc1XU9p1-SQFVvQWYZHEjcDKZ8/edit) and [copy doc](https://docs.google.com/document/d/1qNEjihmySR6l6LZJ-SkSVTpaBSxuY8imoseN0OLkiMw/edit#)

## Issue / Card

Fixes [#4635] (https://github.com/canonical-web-and-design/web-squad/issues/4635)

## Screenshots
<img width="1440" alt="IoT Device Programme | Partners | Canonical ()" src="https://user-images.githubusercontent.com/57550290/144094774-a92af2db-e923-41c3-b610-f821249ed2a0.png">

webteam-app commented:
Demo starting at https://canonical-com-453.demos.haus

anthonydillon commented on templates/partners/iot-device.html:
```diff
@@ -0,0 +1,99 @@
+  </div>
+</section>
+
+{% include 'partial/_iot-device-footer.html' %}
```
Do you need an include if this content is only used once?

anthonydillon commented on templates/partners/iot-device.html:
```diff
@@ -0,0 +1,99 @@
+    <div class="col-6">
+      <p>Our users have a high level of expectation in their experience with Ubuntu. Whether at home, in development, or production, an easy and predictable user experience is important to us. To ensure this success, we partner closely with vendors worldwide to test, validate, and certify their portfolios to ensure this success.</p>
+      <p>Our partners gain access to a worldwide audience looking to purchase devices built to run on Ubuntu by participating in the partner program. The top manufacturers in the world are all engaged with Canonical on promoting their platforms as Ubuntu Certified.</p>
+      <p><a href="https://ubuntu.com/certified" class="p-link--external">Learn more about Ubuntu certified hardware</a></p>
```
Ubuntu Certified Hardware is capitalised everywhere. Can you check which one it is please.

anthonydillon reviewed (changes requested):
Looking good with some small requests

minkyngkm commented:
Thanks @anthonydillon for reviewing! Its been all fixed 🙌

anthonydillon commented on templates/partners/iot-device.html:
```diff
@@ -0,0 +1,99 @@
+  </div>
+</section>
+<section class="p-strip--light is-deep">
+  <div class="u-fixed-width">
```
This is still u-fixed-width

minkyngkm replied:
sorry I updated the other one. Now it's fixed!

anthonydillon commented on templates/partners/iot-device.html:
```diff
@@ -0,0 +1,123 @@
+      <p><a href="https://ubuntu.com/certified" class="p-link--external">Learn more about Ubuntu certified hardware</a></p>
+    </div>
+    <div class="col-6 u-align--center u-hide--small" style="padding:3rem;" >
+      <img src="https://assets.ubuntu.com/v1/706b259a-ubuntu_certified_hex.svg"  style="max-width:10rem;" alt="">
```
Do we use the image template tag on canonical.com? It works be nice to add here or create an issue for it.

minkyngkm replied:
It doesn't work on canonical.com - yes I will make an issue of it!

anthonydillon commented on templates/partners/iot-device.html:
```diff
@@ -0,0 +1,123 @@
+{% block meta_image %}https://assets.ubuntu.com/v1/a491f87c-Ubuntu+Certified+Hardware.svg{% endblock %}
+
+{% block content %}
+<section class="p-strip--suru-bottomed is-deep" style="padding-bottom: 8rem;">
```
Does this need to be deep? The padding bottom adds a lot of space on small screens.

minkyngkm replied:
I think it needs to be deep because it looks like this.
![image](https://user-images.githubusercontent.com/57550290/144206181-44e57c58-8304-4437-9ed7-d33066472882.png)
I removed the `padding-bottom` style instead.

anthonydillon reviewed (changes requested):
Sorry, a more few questions in the code.

minkyngkm commented on templates/partners/iot-device.html:
```diff
@@ -0,0 +1,123 @@
+<section class="p-strip is-deep">
+  <div class="row">
+    <div class="col-12 u-align--center">
+      <h4 class="p-muted-heading" style="max-width: none;">A SELECTION OF OUR IOT PARTNERS</h4>
```
The `p-muted-heading` capitalises the letters. Is it okay to remove the class?

anthonydillon replied:
No sorry, I mean in the mark up we should make it sentence case but the class will convert them up uppercase as styling.

minkyngkm replied:
Ohh sorry! - it's been updated!

spencerbygraves commented:
@minkyngkm a few thoughts from me...

- I don't think it works to have two images in the hero, I would remove the first gateway image and make the other one larger to match the other pages in this section
- The logo section has the wrong title style ("A selection of our IoT partners"), you can see the correct one on the other pages in the Partners section
- The Eurotech logo is small compared to the others. Can it be a bit larger please, and perhaps on the second line so we don't have one logo by itself

Thanks

minkyngkm commented:
Thanks @spencerbygraves for reviewing. I've updated all following your suggestions. Hopefully it looks better now - [Demo](https://canonical-com-453.demos.haus/partners/iot-device)

One question for @anthonydillon
I've replaced the logo section with the new vanilla logo pattern (Refer to [this](https://vanillaframework.io/docs/patterns/logo-section)). But the title gets capitalised by the pattern. I added `text-transform:none` as inline style. What do you think about it? The title on the other pages are all capitalised though.

anthonydillon reviewed (approved):
LGTM :+1: thanks for the changes
